### PR TITLE
Switch to dune and ensure the AST is up-to-date wrt opam-file-format

### PR DESCRIPTION
**Pull request by kit-ty-kate**
This should ensure that new opam-file-format AST contructs are handled and would fail otherwise.

### CODE REVIEW

File: tests/run.sh [hunk 1]
```diff
@@ -1,11 +1,11 @@
-#!/bin/bash
+#!/usr/bin/env bash
 set -ue
 #set -x
 
 echo "## Testing empty stdin"
-diff -u <(./opam-ed "add bin [\"foo\"]" <&-) <(echo "bin: [\"foo\"]")
+diff -u <(opam-ed "add bin [\"foo\"]" <&-) <(echo "bin: [\"foo\"]")
```
Comment (AltGr) [inline tests/run.sh:6]: Hm couldn't it be confusing that the test script tests the installed version of the program and not the locally-built source you're working on ?
Maybe we could do something like `alias opam-ed="dune exec -- opam-ed` ?

**Review by AltGr — APPROVED** [review]
Thanks! Apart from the small bit on the test script, LGTM!

File: tests/run.sh (hunk 1 above)
Comment (kit-ty-kate) [reply tests/run.sh:6]: `dune runtest` includes `_build/install/bin` into the `PATH`, so it's only a problem if it's ran by hand, in which case running `dune exec -- ./run.sh` would do it too. I think it's better to give the options to test either the installed version or the local one.

Comment (AltGr) [reply tests/run.sh:6]: Hm alright this makes sense.

**Pull request merged**